How to build core component (#121)
infinum/frontend-handbook: React/Recipes/How to build core component.md

      mx="auto"
      bg="gray.200"
    />
    <Heading as="h1" size="sm" px="5" mb="4" textAlign="center">
      {`${user.name} ${user.surname}`}
    </Heading>
    <Text fontSize="xs" px="5" mb="5" noOfLines={3} textAlign="center">
      {user.bio}
    </Text>
  </Box>
);
```

```tsx
// ./src/components/shared/user/JobCard/JobCard.tsx

import * as React from "react";
import { Box, Heading, Image, Text } from "@chakra-ui/react";
import { FC } from "react";
import { IJob } from "../../../../interfaces/IJob";

export interface IJobCardProps {
  job: IJob;
}

export const JobCard: FC<IJobCardProps> = ({ job }) => (
  <Box borderRadius="md" border="1px" borderColor="gray.200">
    <Image
      src={job.thumbnail}
      borderTopRadius="md"
      w="100%"
      h="200px"
      fit="cover"
      mb="4"
    />
    <Heading as="h1" size="sm" px="5" mb="4">
      {job.title}
    </Heading>
    <Text fontSize="xs" px="5" pb="5" noOfLines={3}>
      {job.description}
    </Text>
  </Box>
);
```

### When this is OK?

When we are building a small one-off components which doesn't contain a lot of code and  are not shared across components.  

### Why this approach can become a problem?

> This example is not so bad as it seems at firs because we don't have to name components (naming is hard), but as this component grows and new features are added, inline style will become a problem.

Reviewer: ```suggestion
This example is not as bad as it seems at first because we don't have to name components (naming is hard), but as this component grows and new features are added, the inline style will become a problem.
```

      mb: "4",
      textAlign: "center"
    },
    body: {
      textAlign: "center"
    }
  };
};

const variants: MultiStyleConfig<CardAnatomy>["variants"] = {
  solid: variantSolid,
  rounded: variantRounded
};

const defaultProps: MultiStyleConfig<CardAnatomy>["defaultProps"] = {
  variant: "solid"
};

export default {
  parts: parts.keys,
  baseStyle,
  defaultProps,
  variants
} as MultiStyleConfig<CardAnatomy>;
```

After a card theme is done we'll update the theme to include the new Card component style.

```tsx
import { extendTheme } from "@chakra-ui/react";

import Card from "./components/card";

export const theme = extendTheme({
  components: {
    Card
  }
});
```

#### Card component

Now we can implement the actual Card component.

#### Consuming style config

Since the new Card component is not part of Chakra UI we need to create a new React component and consume the style we just created. We can do that using `useMultiStyleConfig` hook.

#### useMultiStyleConfig API

```tsx
const styles = useMultiStyleConfig("Card", props);
```

> [API Docs](https://chakra-ui.com/docs/styled-system/theming/component-style#usemultistyleconfig-api)

#### Using StylesProvider

> In the rood component we can access the whole style config and provide it to all the parts via `StylesProvider` context provider. This way we can for example set the variant only on the root element and all the parts will inherit that variant styles.

Reviewer: ```suggestion
In the root component, we can access the whole style config and provide it to all the parts via `StylesProvider` context provider. This way we can, for example, set the variant only on the root element and all the parts will inherit that variant styles.
```

        {children}
      </chakra.div>
    </StylesProvider>
  );
});

export const CardImage = forwardRef<ImageProps, "img">((props, ref) => {
  const { image } = useStyles();

  return <Image ref={ref} {...image} {...props} />;
});

```

#### Complete example

```tsx
// ./src/components/core/Card/Card.tsx

import {
  chakra,
  forwardRef,
  Heading,
  HeadingProps,
  HTMLChakraProps,
  Image,
  ImageProps,
  StylesProvider,
  Text,
  TextProps,
  ThemingProps,
  useMultiStyleConfig,
  useStyles
} from "@chakra-ui/react";
import { __DEV__ } from "@chakra-ui/utils";

export interface CardOptions {}

export interface CardProps
  extends HTMLChakraProps<"div">,
    CardOptions,
    ThemingProps {}

export const Card = forwardRef<CardProps, "div">((props, ref) => {
  const { children } = props;

  const styles = useMultiStyleConfig("Card", props);

  return (
    <StylesProvider value={styles}>
      <chakra.div ref={ref} __css={styles.card} {...props}>
        {children}
      </chakra.div>
    </StylesProvider>
  );
});

if (__DEV__) {
  Card.displayName = "Card";
}

export const CardImage = forwardRef<ImageProps, "img">((props, ref) => {
  const { image } = useStyles();

>   return <Image ref={ref} __css={image} {...props} />;

Author: ```suggestion
  return <Image ref={ref} {...image} {...props} />;
```
